lms/djangoapps/courseware/views/views.py

```diff
@@ -867,6 +866,11 @@ def _progress(request, course_key, student_id):
     # additional DB lookup (this kills the Progress page in particular).
     student = User.objects.prefetch_related("groups").get(id=student.id)
 
+    if student_id != request.user.id:
```

Comment:
Comparing `student_id` against `student.id` would be better here, since the note above wants us to use `student` in place of `request.user.id`.